src/main/java/org/jenkinsci/plugins/gitclient/CliGitAPIImpl.java

```diff
@@ -1142,13 +1142,21 @@ public void addSubmodule(String remoteURL, String subdir) throws GitException, I
     /**
      * Sync submodule URLs
      *
+     * @param recursive Checkout submodules recursively
      * @throws hudson.plugins.git.GitException if underlying git operation fails.
      * @throws java.lang.InterruptedException if interrupted.
      */
-    public void submoduleSync() throws GitException, InterruptedException {
+    public void submoduleSync(boolean recursive) throws GitException, InterruptedException {
         // Check if git submodule has sync support.
         // Only available in git 1.6.1 and above
-        launchCommand("submodule", "sync");
+        if (recursive)
+        {
+            launchCommand("submodule", "sync", "--recursive");
```

Comment:
Could you please check since which version cli git supports this submodule sync option? Then we may have to discuss if this has to handled here explicitly (ignoring this `--recursive` and log an information about it).

Reply:
Command line git 1.8.3 includes `git submodule sync --recursive`.  The oldest version of git officially supported by the git client plugin is 1.7.10.  All the command line git versions I've tested prior to 1.9 (1.7.1 in CentOS, 1.7.10 on Windows, 1.8.3 on Debian 7, etc.) are known to have various issues when working with the plugin. 

Many of those issues are described in the tests with `isAtLeastVersion` conditionals.

I would be interested to know if git as included in CentOS 6 (git 1.7.1) includes `git submodule sync --recursive`.  If it does, then I'd recommend we not bother with a conditional.  If it doesn't, then we should probably place a conditional around the recursive call in order to avoid breaking CentOS 6 users.  While their command line git version is not officially supported, it continues to work in many useful cases.

Reply:
Git version 1.7.1 (on CentOS 6) supports `git submodule sync` .  All command line git versions supported by the git plugin support it.